src/mlir/dialect/VeronaTypes.cc

```diff
@@ -319,15 +310,13 @@ namespace mlir::verona
             os << "imm";
             break;
         }
-        break;
-      }
-    }
+      });
   }
 
   bool isaVeronaType(Type type)
   {
-    return type.getKind() >= FIRST_VERONA_TYPE &&
-      type.getKind() < LAST_VERONA_TYPE;
+    return type.isa<MeetType>() || type.isa<JoinType>() ||
+      type.isa<IntegerType>() || type.isa<CapabilityType>();
```

Comment:
`isa` takes multiple arguments, so you can do `return type.isa<MeetType, JoinType, IntegerType, CapabilityType>();`.

As a side-note, I think we eventually should have a `VeronaType` class, then we can replace calls to `isaVeronaType(type)` by `type.isa<VeronaType>()`.